gk/main.c

```diff
@@ -616,15 +773,35 @@ add_new_flow_from_policy(
 		return NULL;
 	}
 
-	/* Create a new flow entry. */
+	/*
+	 * We double the hash table size to alleviate memory pressure
+	 * when the table is full.
+	 */
```

Comment:
Same comment as below -- is the point of this patch really be be expanding the table, or do we only need need the timer for reclaiming ("pruning") old entries instead?

Reply:
I see your point. I think the memory pressure is in terms of the flow tables, in which they cannot handle more flow entries once they are full. Could you please help to clarify this? @AltraMayor

Reply:
@cjdoucette does have a point. I expect that Gatekeeper will be carefully configured because it has to stand all attackers can through at it. So we can assume that the hash tables will be set at their maximum, meaning that we won't have memory to keep the old table plus another table that is double that size. We need a way to randomly drop entries under these extreme cases.